apps/prairielearn/src/lib/groups.sql

```diff
@@ -142,10 +142,11 @@ SELECT
   bool_or(gr.can_assign_roles_during_assessment) AS can_assign_roles_during_assessment
 FROM
   group_roles as gr
-  JOIN group_user_roles as gu ON gr.id = gu.group_role_id
+  JOIN group_user_roles as gur ON (gr.id = gur.group_role_id)
 WHERE
   gr.assessment_id = $assessment_id
-  AND gu.user_id = $user_id;
+  -- TODO probably a good idea to add a join with groups before doing this check?
+  AND gur.user_id = $user_id;
```

Comment:
Minor concern with indexes here. Group user roles only has good indices for searches based on the (new) primary key or a group ID. There is no index on user ID or group role ID. I don't know if it's better to add an index here (on role ID?) or to change the query to either include the group ID as a parameter or to find the group ID itself from the assessment instance and user ID. Thoughts? This is a concern regardless of if this PR lands or not.